### PR TITLE
Scan project persistence

### PR DESCRIPTION
**Pull request by penenadpi**
- Assigning projects to persisted scan run results
- API to add generate new project or assign configuration (expanded in future)

### CODE REVIEW

File: examples/projects_example.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import requests
+import sys
+
+
+
```
Comment (anzoman) [inline examples/projects_example.py:5]: Is there too many blank lines?

File: examples/projects_example.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    "configid": "updated_config_value"
+}
+
+
+par["projectid"] = projectid
+
+url = "http://127.0.0.1:8000/set_project_config"
+
+
+
```
Comment (anzoman) [inline examples/projects_example.py:26]: Too many blank lines.

File: examples/test_scanner_simple.py [hunk 3]
```diff
@@ -5,9 +5,10 @@
 
 multipart_form_data = {
     "iac": (sys.argv[1], open(sys.argv[1], "rb")),
-    "checks": (None, sys.argv[2]),
+    "checks": (None, sys.argv[2])
 }
-response = requests.post(URL, files=multipart_form_data)
+par = { "projectid" : (None, "penenadpi") }
```
Comment (anzoman) [inline examples/test_scanner_simple.py:10]: Reformat.

```suggestion
par = { 
    "projectid" : (None, "penenadpi") 
}
```

File: src/iac_scan_runner/api.py [hunk 4]
```diff
@@ -159,7 +159,12 @@ async def post_scan(iac: UploadFile = File(..., description='IaC file (zip or ta
             checks = []
         else:
             checks = list(set(checks[0].split(",")))
-        scan_output = scan_runner.scan_iac(iac, checks, scan_response_type)
+        print("PROJECTID")
+        print(projectid)
```
Comment (anzoman) [inline src/iac_scan_runner/api.py:163]: Remove prints.

File: src/iac_scan_runner/api.py [hunk 5]
```diff
@@ -159,7 +159,12 @@ async def post_scan(iac: UploadFile = File(..., description='IaC file (zip or ta
...
+        if not projectid:
+            projectid=""
```
Comment (anzoman) [inline src/iac_scan_runner/api.py:165]: Reformat:

```suggestion
            projectid = ""
```

Comment (anzoman) [reply src/iac_scan_runner/api.py:165]: Also how does this change the value of `project_id`. Empty string `""` is still considered as `False`.

File: src/iac_scan_runner/api.py [hunk 6]
```diff
@@ -202,3 +207,36 @@ async def delete_scan_result(uuid: str) -> JSONResponse:
             return JSONResponse(status_code=status.HTTP_200_OK, content=f"No such scan result {uuid}")     
     except Exception as e:
         return JSONResponse(status_code=status.HTTP_400_BAD_REQUEST, content=str(e))         
+
+@app.post("/new_project", summary="Retrieve particular scan result by given uuid", responses={200: {}, 400: {"model": str}})
```
Comment (anzoman) [inline src/iac_scan_runner/api.py:211]: Retrieve particular scan result by given uuid? Is that what this API endpoint really does?
Shouldn't it be "Create a new project" or something?

Update docstring and summary.

File: src/iac_scan_runner/api.py [hunk 7]
```diff
@@ -202,3 +207,36 @@ async def delete_scan_result(uuid: str) -> JSONResponse:
...
+        print(creatorid)
+        scan_project = ScanProject() 
+
+        pid=scan_project.new_project(creatorid, "")  
+         
+        return JSONResponse(status_code=status.HTTP_200_OK, content=pid)
+    except Exception as e:
+        return JSONResponse(status_code=status.HTTP_400_BAD_REQUEST, content=str(e))         
+        
+@app.post("/set_project_config", summary="Retrieve particular scan result by given uuid", responses={200: {}, 400: {"model": str}})
```
Comment (anzoman) [inline src/iac_scan_runner/api.py:228]: Update docstring and summary.

File: src/iac_scan_runner/results_summary.py [hunk 8]
```diff
@@ -213,6 +213,18 @@ def dump_outcomes(self, file_name: str):
...
+    def evaluate_verdict(self):
+        verdict = "Passed"
+        for check in self.outcomes:
+            if(check!="uuid" and check!="time" and check!="problems" and check!="passed" and check!="total"):       
+                #print(self.outcomes[check]["status"])
+                if(self.outcomes[check]["status"]=="Problems"):
+                    self.outcomes["verdict"] = "Problems"
+                    return "Problems"
+        self.outcomes["verdict"] = "Passed"
+        return "Passed"
```
Comment (anzoman) [inline src/iac_scan_runner/results_summary.py:225]: Aren't those changes from #19?
If they are this PR should be a draft and there should be a note that we should first reveiw and merge #19 and then proceed with this one.

**Review by anzoman — CHANGES_REQUESTED** [review]

File: examples/projects_example.py (hunk 1 above)
Comment (penenadpi) [reply examples/projects_example.py:5]: I have deleted them

File: examples/projects_example.py (hunk 2 above)
Comment (penenadpi) [reply examples/projects_example.py:26]: Also deleted

File: src/iac_scan_runner/api.py (hunk 4 above)
Comment (penenadpi) [reply src/iac_scan_runner/api.py:163]: Done

File: examples/test_scanner_simple.py (hunk 3 above)
Comment (penenadpi) [reply examples/test_scanner_simple.py:10]: Done

File: src/iac_scan_runner/api.py (hunk 7 above)
Comment (penenadpi) [reply src/iac_scan_runner/api.py:228]: Updated

File: src/iac_scan_runner/results_summary.py (hunk 8 above)
Comment (penenadpi) [reply src/iac_scan_runner/results_summary.py:225]: Yes, I will add the Draft prefix and also resolve #19 . We should merge this after #19

File: src/iac_scan_runner/api.py (hunk 5 above)
Comment (penenadpi) [reply src/iac_scan_runner/api.py:165]: To make sure it is an empty string before inserting into DB

**Review by anzoman — APPROVED** [review]

**Pull request merged**